cmd/can/main.go

```diff
@@ -1,18 +1,87 @@
 package main
 
 import (
+	"bytes"
+	"flag"
 	"fmt"
 	"github.com/sasswart/gin-in-a-can/config"
 	"github.com/sasswart/gin-in-a-can/openapi"
 	"github.com/sasswart/gin-in-a-can/render"
 	golang "github.com/sasswart/gin-in-a-can/render/go"
 	"github.com/sasswart/gin-in-a-can/tree"
+	"io"
 	"os"
+	"path/filepath"
+	"sync"
 )
 
 func main() {
 	fmt.Printf("can %s\n", config.SemVer)
-	cfg := mustLoadConfig()
+
+	if !flag.Parsed() { // Sorts out buggy tests
+		flag.BoolVar(&config.VersionFlagSet, "version", false, "Print Can version and exit")
+		flag.BoolVar(&config.Debug, "debug", false, "Enable debug logging")
+		flag.StringVar(&config.ConfigFilePath, "configFile", ".", "Specify which config file to use")
+		flag.BoolVar(&config.Dryrun, "dryrun", false,
+			"Toggles whether to perform a render without writing to disk."+
+				"This works particularly well in combination with -debug")
+		flag.StringVar(&config.TemplateNameFlag, "template", "", "Specify which template set to use")
+		// TODO should we support stdout as an option here?
+		flag.StringVar(&config.OutputPathFlag, "outputPath", "", "Specify where to write output to")
+		flag.Parse()
+	}
+
+	absCfgPath, err := filepath.Abs(config.ConfigFilePath)
+	if err != nil {
+		fmt.Printf("could not resolve relative config path: %v\n", err)
+		return
+	}
+	config.ConfigFilePath = absCfgPath
+
+	if config.VersionFlagSet {
+		fmt.Printf("Can: v%s\n", config.SemVer)
+		os.Exit(0)
+	}
+
+	if config.Debug {
+		fmt.Printf("[v%s]::Using config file \"%s\".\n", config.SemVer, config.ConfigFilePath)
+	}
+
+	configFileReader, err := os.Open(absCfgPath)
+	if err != nil {
+		fmt.Printf("failed to open config file: %v", err)
```

Comment:
feels like we should os.Exit() here with a non-zero exit code

Reply:
done